vim-patch: update runtime files

clason opened the pull request:
- vim-patch:cd8a3eaf5348
- vim-patch:4f174f0de90b (omit conditional `vim9.vim` file)


We may want to squash-merge these as cd8a3eaf5348 by itself is broken on Neovim?

clason committed:
vim-patch:cd8a3eaf5348

runtime(dist): centralize safe executable check and add vim library (vim/vim#13413)

Follow up to 816fbcc26 (patch 9.0.1833: [security] runtime file fixes,
2023-08-31) and f7ac0ef50 (runtime: don't execute external commands when
loading ftplugins, 2023-09-06).

This puts the logic for safe executable checks in a single place, by introducing
a central vim library, so all filetypes benefit from consistency.

Notable changes:
- dist#vim because the (autoload) namespace for a new runtime support
  library. Supporting functions should get documentation. It might make
  life easier for NeoVim devs to make the documentation a new file
  rather than cram it into existing files, though we may want
  cross-references to it somewhere…
- The gzip and zip plugins need to be opted into by enabling execution
  of those programs (or the global plugin_exec). This needs
  documentation or discussion.
- This fixes a bug in the zig plugin: code setting s:tmp_cwd was removed
  in f7ac0ef50 (runtime: don't execute external commands when loading
  ftplugins, 2023-09-06), but the variable was still referenced. Since
  the new function takes care of that automatically, the variable is no
  longer needed.

https://github.com/vim/vim/commit/cd8a3eaf5348feacfecab4b374b7ea4ce6a97422

Co-authored-by: D. Ben Knoble <ben.knoble+github@gmail.com>

clason committed:
vim-patch:4f174f0de90b

runtime(dist): add legacy version for central vim library

Also, enable the zip and gzip plugins by default, unless those variables
were not explicitly set by the user.

related: vim/vim#13413

https://github.com/vim/vim/commit/4f174f0de90b52937ddaf1e6db98e9731930ff7c

Co-authored-by: Christian Brabandt <cb@256bit.org>

seandewar commented on runtime/autoload/dist/vim.vim:
```diff
@@ -0,0 +1,30 @@
+    endfunction
+else
+    def dist#vim#IsSafeExecutable(filetype: string, executable: string): bool
+      return dist#vim9#IsSafeExecutable(filetype, executable)
```
This line throws E133, as `:return` is still expected to be parsed from inside a `:function` even when evaluation is being skipped over the dead `:else` branch.

The correct way to fix this would be to use `:finish` similar to what's shown in `:h vim9-mix`; I'll open an upstream PR for this.

clason replied:
Thanks, I'll include it!

clason committed:
vim-patch:b2a4c110a5d1

runtime(dist): Make dist/vim.vim work properly when lacking vim9script support (vim/vim#13487)

`:return` cannot be used outside of `:function` (or `:def`) in older Vims
lacking Vim9script support or in Neovim, even when evaluation is being skipped
in the dead `:else` branch.

Instead, use the pattern described in `:h vim9-mix`, which uses `:finish` to end
script processing before it reaches the vim9script stuff.

https://github.com/vim/vim/commit/b2a4c110a5d13bc794f4eddb2e88a4e8fe9dfbea

Co-authored-by: Sean Dewar <seandewar@users.noreply.github.com>
